esmvalcore/iris_helpers.py

```diff
@@ -1,8 +1,73 @@
 """Auxiliary functions for :mod:`iris`."""
+import dask.array as da
 import iris
+import iris.cube
+import iris.util
 import numpy as np
 
 
+def add_leading_dim_to_cube(cube, dim_coord):
+    """Add new leading dimension to cube.
+
+    An input cube with shape ``(x, ..., z)`` will be transformed to a cube with
+    shape ``(w, x, ..., z)`` where ``w`` is the length of ``dim_coord``. Note
+    that the data is broadcasted to the new shape.
```

Comment:
maybe add `dim_coord` needs to be scalar 1-d rather than 2-d etc

Reply:
Great idea! I just added this to the doc and also added a check in the code for this.

Reply:
cool, cheers, bud!